### PR TITLE
Updated `reduce_sum` docs to reflect addition of `lupmf`

### PR DESCRIPTION
**Pull request by bbbales2**
~~I'm leaving this as a draft until the rest of the `lupmf` docs get in (this will need to link to them)~~

Skip that just gonna finish this up and get it updated so it does't get forgotten

### CODE REVIEW

**Comment by bbbales2** [comment]
@wds15 can you review this? It's lupmfs for `reduce_sum`

**Review by rok-cesnovar — COMMENTED** [review]
Maybe mention that in order to use this you need Stan 2.25+?

**Comment by bbbales2** [comment]
@rok-cesnovar good call updating now

File: knitr/reduce-sum/reduce_sum_tutorial.Rmd [hunk 1]
```diff
@@ -149,59 +151,78 @@ statement:
...
+Now it is clear that the calculation is the sum (up to a
+proportionality constant) of a number of conditionally independent
+Bernoulli log probability statements. Whenever we need to calculate
+a large sum where each term is independent of all others and associativity
+holds, then `reduce_sum` is useful.
 
 To use `reduce_sum`, a function must be written that can be used to compute
 arbitrary sections of this sum.
 
+Note we used `binomial_logit_lupmf` instead of `binomial_logit_lpmf`.
```
Comment (rok-cesnovar) [inline knitr/reduce-sum/reduce_sum_tutorial.Rmd:171]: we use or are using? But I am not a native speaker so :shrug:

File: knitr/reduce-sum/reduce_sum_tutorial.Rmd [hunk 2]
```diff
@@ -149,59 +151,78 @@ statement:
...
+This is because we only need this likelihood term up to a proportionality
+constant for MCMC to work and for some distributions this can make code
+run noticeably faster. Because of the way that `_lupmf` features work,
+Stan only allows them in the model block or in user-defined probability
+distribution functions, and for us to use `binomial_logit_lupmf` the
+function we write for `reduce_sum` will need to be a probability distribution
+function (suffixed with `_lpdf` or `_lpmf`).
```
Comment (rok-cesnovar) [inline knitr/reduce-sum/reduce_sum_tutorial.Rmd:178]:
```suggestion
run noticeably faster. There is a bit of catch here. Stan only allows `_lupmf` in the model block
or in user-defined probability distribution functions. Thus, for us to use `binomial_logit_lupmf` the
function we use with `reduce_sum` will need to be a probability distribution
function (suffixed with `_lpdf` or `_lpmf`).
```

Comment (rok-cesnovar) [reply knitr/reduce-sum/reduce_sum_tutorial.Rmd:178]: This is just a suggestion.

File: knitr/reduce-sum/reduce_sum_tutorial.Rmd [hunk 3]
```diff
@@ -149,59 +151,78 @@ statement:
...
+If the difference in the normalized and unnormalized functions is not
+relevant for your application, you can call your `reduce_sum` function
+whatever you like.
```
Comment (rok-cesnovar) [inline knitr/reduce-sum/reduce_sum_tutorial.Rmd:181]:
```suggestion
If the difference in the performance of normalized and unnormalized functions is not
relevant for your application, you can call your `reduce_sum` function
whatever you like.
```

File: knitr/reduce-sum/reduce_sum_tutorial.Rmd [hunk 4]
```diff
@@ -13,7 +13,9 @@ This introduction to `reduce_sum` copies directly from Richard McElreath's
 
 ## Introduction
 
-Stan 2.23 introduced `reduce_sum`, a new way to parallelize the execution of
+**Note:** This has been rewritten to use `_lupmf` which requires Cmdstan 2.25.
```
Comment (rok-cesnovar) [inline knitr/reduce-sum/reduce_sum_tutorial.Rmd:16]:
```suggestion
**Note:** This has been rewritten to use unnormalized distribution functions ( `_lupdf˙/˙_lupmf`) which requires Cmdstan 2.25 or newer.
```

**Review by rok-cesnovar — CHANGES_REQUESTED** [review]
Minor review comments.

**Comment by bbbales2** [comment]
@rok-cesnovar wow took me a long time to get around to this. I think all your edits look good so I did them

**Review by rok-cesnovar — APPROVED** [review]
Looks good!

**Pull request merged**